### PR TITLE
feat: adding price scaling for sudoswap xyk pool

### PR DESCRIPTION
**Pull request by JackShort**
* This scales the prices on the frontend to handle sudoswap xyk pool
* good collection to test: White Hearts 1.0

### CODE REVIEW

File: src/nft/utils/pooledAssets.ts [hunk 1]
```diff
@@ -44,12 +45,33 @@ export const calcSudoSwapPrice = (asset: GenieAsset, position = 0): string | und
   let currentPrice = BigNumber.from(sudoSwapPool.spotPrice)
   const delta = BigNumber.from(sudoSwapPool.delta)
   const poolFee = BigNumber.from(sudoSwapPool.fee)
+  let virtualTokenBalance = BigNumber.from(sudoSwapPool.spotPrice)
+  let virtualNFTBalance = BigNumber.from(sudoSwapPool.delta)
 
-  for (let i = 0; i <= position; i++) {
+  if (sudoSwapPool.bondingCurve === BondingCurve.Xyk) {
```
Comment (cbachmeier) [inline src/nft/utils/pooledAssets.ts:51]: this function has gotten rather large, do we want to split it into smaller functions for each curve?

**Review by cbachmeier — CHANGES_REQUESTED** [review]
Functionality works well. We should still delete the function `isNotXYKPool` in `nft/utils`. Also gave a organization suggestion. 

One other non blocking idea where some additional logic could be simplified if `calcPoolPrice` checked if iasset was in a sudoswap pool and routed to the appropriate helper function. Basically allowing that function to handle all routing.

File: src/nft/utils/pooledAssets.ts [hunk 2]
```diff
@@ -32,6 +33,50 @@ const calculateScaledPrice = (currentPrice: BigNumber, poolFee: BigNumber): BigN
...
+  sudoSwapPool: Pool,
+  position = 0
+): BigNumber | undefined => {
+  let virtualTokenBalance = BigNumber.from(sudoSwapPool.spotPrice)
+  let virtualNFTBalance = BigNumber.from(sudoSwapPool.delta)
+
+  if (virtualNFTBalance.sub(BigNumber.from(1)).gt(BigNumber.from(0))) {
+    currentPrice = virtualTokenBalance.div(virtualNFTBalance.sub(BigNumber.from(1)))
+  } else {
+    return undefined
```
Comment (cbachmeier) [inline src/nft/utils/pooledAssets.ts:63]: I think this undefined might be causing a crash when trying to add past the available amount. Should we return 0 instead?
xyk

File: src/nft/utils/pooledAssets.ts [hunk 3]
```diff
@@ -120,6 +171,13 @@ export const calcPoolPrice = (asset: GenieAsset, position = 0) => {
   return price.toString()
 }
 
+// TODO: a lot of the below typecasting logic can be simplified when GraphQL migration is complete
```
Comment (cbachmeier) [inline src/nft/utils/pooledAssets.ts:174]: my bad, this comment can be removed

File: src/nft/utils/pooledAssets.ts [hunk 4]
```diff
@@ -45,19 +90,25 @@ export const calcSudoSwapPrice = (asset: GenieAsset, position = 0): string | und
   const delta = BigNumber.from(sudoSwapPool.delta)
   const poolFee = BigNumber.from(sudoSwapPool.fee)
 
-  for (let i = 0; i <= position; i++) {
-    if (sudoSwapPool.bondingCurve === BondingCurve.Linear) {
-      currentPrice = currentPrice.add(delta)
-    } else if (sudoSwapPool.bondingCurve === BondingCurve.Exponential) {
-      currentPrice = currentPrice.mul(delta).div(BigNumber.from(PRECISION))
+  if (sudoSwapPool.bondingCurve === BondingCurve.Linear) {
```
Comment (cbachmeier) [inline src/nft/utils/pooledAssets.ts:93]: v clean

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: src/nft/utils/pooledAssets.ts (hunk 2 above)
Comment (JackShort) [reply src/nft/utils/pooledAssets.ts:63]: its actually returning `0` after the refactor

**Review by cbachmeier — APPROVED** [review]
LGTM

**Pull request merged**